### DISCUSSION TITLE
Getting `Type ...XXXProps is not assignable to type React.ReactNode` from `<Modal {...props}>`

### DISCUSSION THREAD

**Discussion by ShookLyngs** [mantinedev/mantine#3976] [Q&A]
When I try to wrap a mantine component like this:
```tsx
import { Modal, ModalProps } from '@mantine/core';

export interface WrappedModalProps {
  someProp: number;
}

export function WrappedModal(props: WrappedModalProps & ModalProps) {
  return (
    <Modal {...props}>...</Modal>
  );
}
```

I'll get this error (from WebStorm):
```
Type ...WrappedModalProps & ModalProps is not assignable to type React.ReactNode
```

I don't understand how the `props` here is related to the `React.ReactNode` type.
Does anyone know what is causing this error?

**Comment by rtivital** [comment]
Try reproducing the issue on codesandbox. Template – https://codesandbox.io/s/mantine-template-9ze89?file=/src/App.tsx:0-6
